packages/core/src/tracing/cls.js

```diff
@@ -282,9 +283,15 @@ function startSpan(spanName, kind, traceId, parentSpanId, w3cTraceContext) {
 
   // If the client code has specified a trace ID/parent ID, use the provided IDs.
   if (traceId) {
-    span.t = traceId;
+    // The incoming trace ID/span ID from an upstream tracer could be shorter than the standard length. Some of our code
+    // (in particular, the binary Kafka trace correlation header X_INSTANA_C) assumes the standard length. We normalize
+    // both IDs here by left-padding with 0 characters.
+
+    // Maintenance note (128-bit-trace-ids): When we switch to 128 bit trace IDs, we need to left-pad the trace ID to 32
```

Comment:
Not directly related to the issue at hand, but I added a couple of those comments with a common identifier that can be searched for easily (128-bit-trace-ids), in all places I could think of that need to be updated when we switch from 64 bit trace IDs to 128 bit trace IDs.